unit_test.sh

```diff
@@ -0,0 +1,135 @@
+  -e POSTGRESQL_SERVICE_HOST="$DATABASE_HOST" \
+  -e POSTGRESQL_USER="$DATABASE_USER" \
+  -e POSTGRESQL_PASSWORD="$DATABASE_PASSWORD" \
+  -e POSTGRESQL_DATABASE="$DATABASE_NAME" \
+  -e POSTGRESQL_PORT="$DATABASE_PORT" \
+  -e RAILS_ENV=test \
+  "$IMAGE:$IMAGE_TAG" \
+  /bin/bash -c 'sleep infinity' || echo "0")
+
+if [[ "$TEST_CONTAINER_ID" == "0" ]]; then
+  echo "Failed to start test container"
+  exit 1
+fi
+
+WORKSPACE=${WORKSPACE:-./}
+ARTIFACTS_DIR="$WORKSPACE/artifacts"
+mkdir -p "$ARTIFACTS_DIR"
+
+# gem install
+echo '===================================='
+echo '=== Installing Gem Dependencies ===='
+echo '===================================='
+set +e
+docker exec "$TEST_CONTAINER_ID" /bin/bash -c 'bundle install --with test'
+TEST_RESULT=$?
+set -e
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '==== ✖ ERROR: GEM INSTALL FAILED ===='
+  echo '====================================='
+  exit 1
+fi
+
+# setup db
+echo '===================================='
+echo '===     Setting Up Database     ===='
+echo '===================================='
+set +e
+docker exec "$TEST_CONTAINER_ID" /bin/bash -c 'bundle exec rake db:test:prepare'
+TEST_RESULT=$?
+set -e
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '====  ✖ ERROR: DB SETUP FAILED   ===='
+  echo '====================================='
+  exit 1
+fi
+
+# setup cyndi
+echo '===================================='
+echo '===       Setting Up Cyndi      ===='
+echo '===================================='
+set +e
+docker cp "$TEST_CONTAINER_ID":/opt/app-root/src/db/cyndi_setup_test.sql "$WORKSPACE/"
+docker cp "$WORKSPACE/cyndi_setup_test.sql" "$DB_CONTAINER_ID":/var/lib/pgsql/
+rm "$WORKSPACE/cyndi_setup_test.sql"
+docker exec "$DB_CONTAINER_ID" /bin/bash -c 'psql -d $POSTGRESQL_DATABASE < cyndi_setup_test.sql'
+TEST_RESULT=$?
+set -e
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '==== ✖ ERROR: CYNDI SETUP FAILED ===='
+  echo '====================================='
+  exit 1
+fi
+
+# unit tests
+echo '===================================='
+echo '===     Running Unit Tests      ===='
+echo '===================================='
+set +e
+docker exec "$TEST_CONTAINER_ID" /bin/bash -c 'bundle exec rake test:validate'
+TEST_RESULT=$?
+set -e
+# Copy test reports
+docker cp "$TEST_CONTAINER_ID":/opt/app-root/src/test/reports/. "$WORKSPACE"/artifacts
+# Prefix name of reports with 'junit-' so jenkins analysis picks them up
+cd "$WORKSPACE/artifacts"
+for FILENAME in TEST-*.xml; do mv "$FILENAME" "junit-$FILENAME"; done
+cd -
+if [[ $TEST_RESULT -ne 0 ]]; then
+  echo '====================================='
+  echo '====  ✖ ERROR: UNIT TEST FAILED  ===='
+  echo '====================================='
+  exit 1
+fi
+
+echo '====================================='
+echo '====   ✔ SUCCESS: PASSED TESTS   ===='
+echo '====================================='
+exit 0
```

Comment:
Remove this 'exit 0', otherwise once unit tests pass the entire pr_check script will exit (discovered that problem yesterday)